loopy/tools.py

```diff
@@ -601,4 +603,82 @@ def is_interned(s):
 def intern_frozenset_of_ids(fs):
     return frozenset(intern(s) for s in fs)
 
+
+# {{{ resolve_name
+
+# https://github.com/python/cpython/commit/1ed61617a4a6632905ad6a0b440cd2cafb8b6414
+
+_DOTTED_WORDS = r"[a-z_]\w*(\.[a-z_]\w*)*"
+_NAME_PATTERN = re.compile(f"^({_DOTTED_WORDS})(:({_DOTTED_WORDS})?)?$", re.I)
+del _DOTTED_WORDS
+
+
+def resolve_name(name):
```

Comment:
Any reason this functionality is not a part of https://github.com/inducer/pytools/pull/57?

Reply:
That's a very good point. Want to steal it from here and then change this (or have me change it)?

Reply:
Or, actually... let me just add it to pytools outright.

Reply:
https://github.com/inducer/pytools/pull/64

Reply:
Thanks!